procrastinate/retry.py

```diff
@@ -17,15 +17,17 @@ class BaseRetryStrategy:
     Children classes only need to implement `get_schedule_in`.
     """
 
-    def get_retry_exception(self, attempts: int) -> Optional[exceptions.JobRetry]:
-        schedule_in = self.get_schedule_in(attempts=attempts)
+    def get_retry_exception(
+        self, exception: Exception, attempts: int
+    ) -> Optional[exceptions.JobRetry]:
+        schedule_in = self.get_schedule_in(exception=exception, attempts=attempts)
         if schedule_in is None:
             return None
 
         schedule_at = pendulum.now("UTC").add(seconds=schedule_in)
         return exceptions.JobRetry(schedule_at)
 
-    def get_schedule_in(self, attempts: int) -> Optional[float]:
+    def get_schedule_in(self, exception: Exception, attempts: int) -> Optional[float]:
```

Comment:
We're giving the exception here because why not? I don't mind it, just making sure I'm not missing anything

Reply:
We'll be doing calls with "exception" here, because the concrete implementation needs to look at the exception to make the call whether it needs to retry or not. Changes in the concrete implementation imply changes in the abstract one too, ohterwise:
- mypy complains
- people will implement it wrongly.

Reply:
Right, I missed that we actually use it in the implementation (obviously.) My bad, thanks for the explanation :+1: